doc/source/whatsnew/v0.25.0.rst

```diff
@@ -665,6 +665,7 @@ Indexing
 ^^^^^^^^
 
 - Improved exception message when calling :meth:`DataFrame.iloc` with a list of non-numeric objects (:issue:`25753`).
+- Improved exception message when calling ``.iloc`` or ``.loc`` with a boolean indexer with different length (:issue:`26658`).
```

Comment:
This isn't quite accurate, since previously there may not have even been an exception, right?

Reply:
Hmm, maybe I misunderstood what @jreback  said above. I thought he was referencing the fact that an exception was already getting raised if there are too many true values (that's the example you provided in the description). Now it's just getting raised whenever the lengths are different.

Reply:
#26658 was initially about too few indexers not raising

```python
In [1]: import pandas as pd
s =
In [2]: s = pd.Series([1, 2, 3])

In [3]: s.iloc[[True, False]]
Out[3]:
0    1
dtype: int64
```